tests/kernel/workq/work/src/main.c

```diff
@@ -175,7 +175,7 @@ static void test_work_init(void)
 {
 	static K_WORK_DEFINE(fnstat, counter_handler);
 
-	struct k_work stack;
+	static struct k_work stack;
```

Comment:
The variable name should change since it isn't on the stack anymore.  Also this isn't a bug in itself, since the work item isn't used in anything other than the init function, so it could be left alone, perhaps with a comment just noting that work items can't be on the stack in non-coherent systems.